// @novu/shared/dist/cjs/types/auth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NONE_AUTH_SCHEME = exports.PassportStrategyEnum = exports.ApiAuthSchemeEnum = exports.SignUpOriginEnum = void 0;
var SignUpOriginEnum;
(function (SignUpOriginEnum) {
    SignUpOriginEnum["WEB"] = "web";
    SignUpOriginEnum["CLI"] = "cli";
    SignUpOriginEnum["VERCEL"] = "vercel";
})(SignUpOriginEnum || (exports.SignUpOriginEnum = SignUpOriginEnum = {}));
var ApiAuthSchemeEnum;
(function (ApiAuthSchemeEnum) {
    ApiAuthSchemeEnum["BEARER"] = "Bearer";
    ApiAuthSchemeEnum["API_KEY"] = "ApiKey";
})(ApiAuthSchemeEnum || (exports.ApiAuthSchemeEnum = ApiAuthSchemeEnum = {}));
var PassportStrategyEnum;
(function (PassportStrategyEnum) {
    PassportStrategyEnum["JWT"] = "jwt";
    PassportStrategyEnum["JWT_CLERK"] = "jwt-clerk";
    PassportStrategyEnum["HEADER_API_KEY"] = "headerapikey";
})(PassportStrategyEnum || (exports.PassportStrategyEnum = PassportStrategyEnum = {}));
exports.NONE_AUTH_SCHEME = 'None';
