// @novu/shared/dist/cjs/types/feature-flags.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.invalidSubjectFlag = exports.invalidSuffixFlag = exports.invalidPrefixFlag = exports.validFlag = void 0;
const vitest_1 = require("vitest");
const feature_flags_1 = require("./feature-flags");
(0, vitest_1.describe)('Flags', () => {
    (0, vitest_1.it)('tests the Typescript compiler errors below', () => {
        (0, vitest_1.expect)(true).toEqual(true);
    });
});
exports.validFlag = 'IS_SOMETHING_ENABLED';
exports.invalidPrefixFlag = 'SOMETHING_ENABLED';
exports.invalidSuffixFlag = 'IS_SOMETHING';
exports.invalidSubjectFlag = 'IS_something_ENABLED';
var ValidFlagsEnum;
(function (ValidFlagsEnum) {
    ValidFlagsEnum["IS_SOMETHING_ENABLED"] = "IS_SOMETHING_ENABLED";
    ValidFlagsEnum["IS_SOMETHING_ELSE_ENABLED"] = "IS_SOMETHING_ELSE_ENABLED";
})(ValidFlagsEnum || (ValidFlagsEnum = {}));
(0, feature_flags_1.testFlagEnumValidity)(ValidFlagsEnum);
var InvalidFlagsEnum;
(function (InvalidFlagsEnum) {
    InvalidFlagsEnum["INVALID_ENABLED"] = "INVALID_ENABLED";
})(InvalidFlagsEnum || (InvalidFlagsEnum = {}));
(0, feature_flags_1.testFlagEnumValidity)(InvalidFlagsEnum);
var NonMatchingKeyValueEnum;
(function (NonMatchingKeyValueEnum) {
    NonMatchingKeyValueEnum["IS_SOMETHING_ENABLED"] = "IS_SOMETHING_ELSE_ENABLED";
})(NonMatchingKeyValueEnum || (NonMatchingKeyValueEnum = {}));
const validateNonMatchingKeyValueEnum = NonMatchingKeyValueEnum;
const validateFeatureFlagsKeysEnum = feature_flags_1.FeatureFlagsKeysEnum;
(0, feature_flags_1.testFlagEnumValidity)(feature_flags_1.FeatureFlagsKeysEnum);
const validateSystemCriticalFlagsEnum = feature_flags_1.SystemCriticalFlagsEnum;
(0, feature_flags_1.testFlagEnumValidity)(feature_flags_1.SystemCriticalFlagsEnum);
