// @novu/shared/dist/cjs/types/message-templates.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageActionStatusEnum = exports.ButtonTypeEnum = exports.TextAlignEnum = exports.EmailBlockTypeEnum = void 0;
var EmailBlockTypeEnum;
(function (EmailBlockTypeEnum) {
    EmailBlockTypeEnum["BUTTON"] = "button";
    EmailBlockTypeEnum["TEXT"] = "text";
})(EmailBlockTypeEnum || (exports.EmailBlockTypeEnum = EmailBlockTypeEnum = {}));
var TextAlignEnum;
(function (TextAlignEnum) {
    TextAlignEnum["CENTER"] = "center";
    TextAlignEnum["LEFT"] = "left";
    TextAlignEnum["RIGHT"] = "right";
})(TextAlignEnum || (exports.TextAlignEnum = TextAlignEnum = {}));
var ButtonTypeEnum;
(function (ButtonTypeEnum) {
    ButtonTypeEnum["PRIMARY"] = "primary";
    ButtonTypeEnum["SECONDARY"] = "secondary";
})(ButtonTypeEnum || (exports.ButtonTypeEnum = ButtonTypeEnum = {}));
var MessageActionStatusEnum;
(function (MessageActionStatusEnum) {
    MessageActionStatusEnum["PENDING"] = "pending";
    MessageActionStatusEnum["DONE"] = "done";
})(MessageActionStatusEnum || (exports.MessageActionStatusEnum = MessageActionStatusEnum = {}));
