// @novu/shared/dist/cjs/types/rate-limiting.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiRateLimitCategoryEnum = exports.ApiRateLimitCostEnum = exports.ApiRateLimitConfigEnum = exports.IApiRateLimitAlgorithm = exports.ApiRateLimitAlgorithmEnum = void 0;
var ApiRateLimitAlgorithmEnum;
(function (ApiRateLimitAlgorithmEnum) {
    ApiRateLimitAlgorithmEnum["BURST_ALLOWANCE"] = "burst_allowance";
    ApiRateLimitAlgorithmEnum["WINDOW_DURATION"] = "window_duration";
})(ApiRateLimitAlgorithmEnum || (exports.ApiRateLimitAlgorithmEnum = ApiRateLimitAlgorithmEnum = {}));
class IApiRateLimitAlgorithm {
}
exports.IApiRateLimitAlgorithm = IApiRateLimitAlgorithm;
ApiRateLimitAlgorithmEnum.BURST_ALLOWANCE, ApiRateLimitAlgorithmEnum.WINDOW_DURATION;
var ApiRateLimitConfigEnum;
(function (ApiRateLimitConfigEnum) {
    ApiRateLimitConfigEnum["ALGORITHM"] = "algorithm";
    ApiRateLimitConfigEnum["COST"] = "cost";
    ApiRateLimitConfigEnum["MAXIMUM"] = "maximum";
})(ApiRateLimitConfigEnum || (exports.ApiRateLimitConfigEnum = ApiRateLimitConfigEnum = {}));
var ApiRateLimitCostEnum;
(function (ApiRateLimitCostEnum) {
    ApiRateLimitCostEnum["SINGLE"] = "single";
    ApiRateLimitCostEnum["BULK"] = "bulk";
})(ApiRateLimitCostEnum || (exports.ApiRateLimitCostEnum = ApiRateLimitCostEnum = {}));
var ApiRateLimitCategoryEnum;
(function (ApiRateLimitCategoryEnum) {
    ApiRateLimitCategoryEnum["TRIGGER"] = "trigger";
    ApiRateLimitCategoryEnum["CONFIGURATION"] = "configuration";
    ApiRateLimitCategoryEnum["GLOBAL"] = "global";
})(ApiRateLimitCategoryEnum || (exports.ApiRateLimitCategoryEnum = ApiRateLimitCategoryEnum = {}));
